src/main/java/seedu/address/model/group/SuperGroup.java

```diff
@@ -32,15 +32,38 @@ public String getName() {
      */
     public void addSubGroup(SubGroup subGroup) {
         requireNonNull(subGroup);
-        subGroups.add(subGroup.name);
+        subGroups.add(subGroup);
+    }
+
+    public UniqueList<SubGroup> getSubGroups() {
+        return subGroups;
+    }
+
+    /**
+     * Finds the SubGroup given the SubGroup name.
+     * @param name the Name of the SubGroup.
+     * @return SubGroup
+     */
+    public SubGroup findSubGroup(String name) {
+        for (SubGroup subGroup: subGroups) {
+            if (subGroup.name.equals(name)) {
+                return subGroup;
+            }
+        }
+        return null;
     }
```

Comment:
Imo much faster/easier with hashmaps

Reply:
I think can go with hashmap. I will update this once UniqueList is updated